### PR TITLE
CDAP-7213, 7774 New Security Settings

### PR DESCRIPTION
**Pull request by awholegunch**
- Adds a new option to "doc-cdap-default.py" to generate the deprecated properties
- "doc-cdap-default.py" checks if the property has been removed from cdap-defaults.xml.
- Adds new "internal security" page. 
- Adjusts existing tables for new SSL settings.
- Adjusts existing tables to replace "default values" with "values" that are the suggested values required to enable the feature.
- **Changed:** Tables in Admin Security settings no longer say "Default Value" (unless it is listing a default value, such as *Ports*) but instead just say "Value", indicating that this is value to be entered for that setting.

Running as a Quick Build: http://builds.cask.co/browse/CDAP-DQB188-14 (passed)

Pages of interest:
- Appendix: cdap-site.xml: http://builds.cask.co/artifact/CDAP-DQB188/shared/build-14/Docs-HTML/4.0.0-SNAPSHOT/en/admin-manual/appendices/cdap-site.html#appendix-cdap-site-xml
- Deprecated Properties: http://builds.cask.co/artifact/CDAP-DQB188/shared/build-14/Docs-HTML/4.0.0-SNAPSHOT/en/admin-manual/appendices/cdap-site.html#deprecated-properties
- Perimeter Security:  http://builds.cask.co/artifact/CDAP-DQB188/shared/build-14/Docs-HTML/4.0.0-SNAPSHOT/en/admin-manual/security/perimeter-security.html#admin-perimeter-security
- Internal Security (**new!**): http://builds.cask.co/artifact/CDAP-DQB188/shared/build-14/Docs-HTML/4.0.0-SNAPSHOT/en/admin-manual/security/internal-security.html
- Client Authentication: http://builds.cask.co/artifact/CDAP-DQB188/shared/build-14/Docs-HTML/4.0.0-SNAPSHOT/en/developers-manual/security/client-authentication.html#obtaining-an-access-token

Fix for: 
- https://issues.cask.co/browse/CDAP-7213
- https://issues.cask.co/browse/CDAP-7774

### CODE REVIEW

**Comment by awholegunch** [comment]
The deprecated XML needs to be resolved with this list: https://github.com/caskdata/cdap/blob/release/4.0/cdap-common/src/main/java/co/cask/cdap/common/conf/Configuration.java#L232

File: cdap-common/src/main/resources/cdap-default.xml [hunk 1]
```diff
@@ -1685,14 +1685,6 @@
   </property>
 
   <property>
-    <name>router.ssl.webapp.bind.port</name>
```
Comment (Nishith) [inline cdap-common/src/main/resources/cdap-default.xml]: I see this being used in code. So, we should confirm before removing them.

File: cdap-common/src/main/resources/cdap-default.xml [hunk 2]
```diff
@@ -1704,23 +1696,6 @@
     </description>
   </property>
 
-  <property>
```
Comment (Nishith) [inline cdap-common/src/main/resources/cdap-default.xml]: I see this being used in code. So, we should confirm before removing them.

File: cdap-common/src/main/resources/cdap-default.xml [hunk 3]
```diff
@@ -1704,23 +1696,6 @@
...
-    <name>router.webapp.bind.port</name>
-    <value>20000</value>
-    <description>
-      CDAP Router service bind port for webapp (deprecated)
-    </description>
-  </property>
-
-  <property>
-    <name>router.webapp.enabled</name>
```
Comment (Nishith) [inline cdap-common/src/main/resources/cdap-default.xml]: I see this being used in code. So, we should confirm before removing them.

**Review by Nishith — COMMENTED** [review]
I only looked at the ssl related things. Someone else could probably review the scripts.

File: cdap-common/src/main/resources/cdap-default.xml (hunk 2 above)
Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml]: ``router.webapp.bind.port``

**Comment by awholegunch** [comment]
Blocked until https://github.com/caskdata/cdap/pull/7418 is merged.

File: cdap-common/src/main/resources/cdap-default.xml (hunk 1 above)
Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml]: Fixed in https://github.com/caskdata/cdap/pull/7418

File: cdap-common/src/main/resources/cdap-default.xml (hunk 2 above)
Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml]: Fixed in https://github.com/caskdata/cdap/pull/7418

File: cdap-common/src/main/resources/cdap-default.xml (hunk 3 above)
Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml]: Fixed in https://github.com/caskdata/cdap/pull/7418

File: cdap-docs/admin-manual/source/security/internal-security.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2016 Cask Data, Inc.
+
+.. _admin-internal-security:
+
+=================
+Internal Security
```
Comment (sreevatsanraman) [inline cdap-docs/admin-manual/source/security/internal-security.rst]: Internal Security is mis-leading. Why don't we call it ```Enabling SSL for System Services```

File: cdap-docs/admin-manual/source/security/internal-security.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+=================
+
+Internal security governs the transmission of data between the :ref:`different components
+of CDAP <admin-manual-cdap-components>`. In CDAP 4.0, SSL can be enabled between the CDAP
+Router and CDAP Master (or App Fabric) components.
+
+To enable external security, see the section on :ref:`Perimeter Security <admin-perimeter-security>`.
```
Comment (sreevatsanraman) [inline cdap-docs/admin-manual/source/security/internal-security.rst]: We shouldn't call it external security. Just call it perimeter security

File: cdap-docs/admin-manual/source/security/internal-security.rst [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+.. _enabling-internal-security:
+
+Enabling Internal Security
```
Comment (sreevatsanraman) [inline cdap-docs/admin-manual/source/security/internal-security.rst]: Enabling SSL for master service

File: cdap-docs/admin-manual/source/security/internal-security.rst (hunk 5 above)
Comment (awholegunch) [reply cdap-docs/admin-manual/source/security/internal-security.rst]: fixed.

File: cdap-docs/admin-manual/source/security/internal-security.rst (hunk 4 above)
Comment (awholegunch) [reply cdap-docs/admin-manual/source/security/internal-security.rst]: fixed.

**Comment by awholegunch** [comment]
Running as a Quick Build: http://builds.cask.co/browse/CDAP-DQB188-15

Fails, but unrelated to this PR. Build will be fixed in another PR.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**